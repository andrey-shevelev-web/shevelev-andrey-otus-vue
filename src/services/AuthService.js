import Cookies from 'js-cookie';

const TOKEN_PARAM_NAME = 'main';
const TOKEN_EXPIRES = 1;

function setToken(token) {
  Cookies.set(TOKEN_PARAM_NAME, token, { expires: TOKEN_EXPIRES });
}

function getToken() {
  return Cookies.get(TOKEN_PARAM_NAME);
}

function removeToken() {
  Cookies.remove(TOKEN_PARAM_NAME);
}

export function authSignin(username, password) {
  let credentials = [username, password].join(':'),
    accessToken = `Basic ${btoa(credentials)}`;

  setToken(accessToken);
}

export function isUserAuthenticated() {
  return !!getToken();
}

export function authSignout() {
  removeToken();
}
