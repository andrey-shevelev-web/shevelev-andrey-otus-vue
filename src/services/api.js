import axios from 'axios';

const axiosInstance = axios.create({
  baseURL: 'https://fakestoreapi.com',
  timeout: 180_000
});

export default axiosInstance;
