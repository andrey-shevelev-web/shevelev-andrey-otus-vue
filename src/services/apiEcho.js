import axios from 'axios';

const axiosInstance = axios.create({
  baseURL: 'https://httpbin.org',
  timeout: 180_000
});

export default axiosInstance;
