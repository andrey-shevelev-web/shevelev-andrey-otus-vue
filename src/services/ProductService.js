import { jsonProducts } from '@/services/JsonProducts';
import api from '@/services/api';
import apiEcho from '@/services/apiEcho';

export function getProductsFromJson() {
  try {
    return JSON.parse(jsonProducts);
  } catch (error) {
    return {};
  }
}

export async function getProducts() {
  const response = await api.get('/products');
  return response.data;
}

export async function postOrderForm(formData) {
  const response = await apiEcho.post('/post', {
    params: { formData }
  });
  return response.data;
}
