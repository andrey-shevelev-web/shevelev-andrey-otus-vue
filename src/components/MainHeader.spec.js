import { describe, it, expect } from "vitest";
import { mount } from "@vue/test-utils";
import { jsonProducts } from "@/services/JsonProducts";
import * as ProductService from "@/services/ProductService";
import { setActivePinia, createPinia } from "pinia";

import component from "@/components/MainHeader.vue";
vi.mock("@/services/ProductService");
vi.mock("primevue/usetoast");

import * as toast from "primevue/usetoast";

describe("MainHeader", () => {
  const mockProducts = JSON.parse(jsonProducts);
  beforeEach(() => {
    toast.useToast = vi.fn().mockImplementation(() => ({ add: vi.fn() }));

    setActivePinia(createPinia());
    ProductService.getProducts = vi.fn().mockResolvedValue(mockProducts);
  });

  it("mounts without error", () => {
    const wrapper = mount(component, {
      global: {
        stubs: ["Toast", "router-link"],
      },
    });
    expect(wrapper.exists()).toBe(true);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
