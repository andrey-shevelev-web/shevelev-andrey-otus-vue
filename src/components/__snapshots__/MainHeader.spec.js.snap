// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MainHeader > mounts without error 1`] = `
"<div class="mb-2 p-2 bg-primary text-white">
  <div class="flex align-items-center justify-content-between flex-wrap px-3">
    <div class="flex align-items-center">
      <div class="text-2xl font-medium text-white">DEMO SHOP</div>
    </div>
    <div class="flex align-items-center">
      <router-link-stub to="[object Object]" class="mr-4 no-underline text-white transition-colors transition-ease-in transition-duration-300 hover:text-yellow-300"></router-link-stub><input class="p-inputtext p-component p-inputtext-sm mr-4" data-pc-name="inputtext" data-pc-section="root" placeholder="Search by name or price">
      <router-link-stub to="[object Object]" class="mr-4 no-underline text-white transition-colors transition-ease-in transition-duration-300 hover:text-yellow-300"></router-link-stub>
      <div class="text-center mr-4 transition-colors transition-ease-in transition-duration-300 hover:text-yellow-300"><i class="pi pi-fw pi-user mr-2"></i><span></span></div>
      <router-link-stub to="{ name: 'Signin' }" class="no-underline text-white transition-colors transition-ease-in transition-duration-300 hover:text-yellow-300"></router-link-stub>
    </div>
  </div>
</div>"
`;
